tensorflow/contrib/go/session.go

```diff
@@ -0,0 +1,147 @@
+package tensorflow
+
+import (
+	"fmt"
+	"runtime"
+
+	"github.com/golang/protobuf/proto"
+)
+
+// Session A Session instance lets a caller drive a TensorFlow graph
+// computation.
+type Session struct {
+	ops     TF_SessionOptions
+	session TF_Session
+	status  TF_Status
+	graph   *Graph
+}
+
+// NewSession initializes a new TensorFlow session.
+func NewSession() (s *Session, err error) {
+	status := TF_NewStatus()
+	ops := TF_NewSessionOptions()
+	s = &Session{
+		ops:    ops,
+		status: status,
+		session: TF_NewSession(
+			ops,
+			status,
+		),
+	}
+
+	err = s.statusToError(status)
+
+	// Release the C allocated memory when the instance is destroyed
+	runtime.SetFinalizer(s, (*Session).FreeAllocMem)
+
+	return
+}
+
+// Run Runs the operations on the target nodes, or all the operations if not
```

Comment:
s/Runs/runs/
s/if not the targets are specified/if the targets are not specified/
s/the Parameter Input in/The parameter input is/